### PR TITLE
SDK-121 [Node.js SDK] Improve error messages 

### PR DESCRIPTION
**Pull request by igor-alekseev-dev**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-121](https://incountry.atlassian.net/browse/SDK-121)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -21,12 +21,15 @@ import {
...
 import { CustomEncryptionConfig } from './validation/custom-encryption-configs';
-import { validate } from './validation/validateDecorator';
+import { validate } from './validation/validate-decorator';
 import { LoggerIO } from './validation/logger';
 import { AuthClient, getApiKeyAuthClient, OAuthClient } from './auth-client';
+import { normalizeErrors } from './normalize-errors-decorator';
 
 const FIND_LIMIT = 100;
 
+const STORAGE_METHOD_ERROR = 'Error during Storage.';
```
Comment (artemgolozubov) [inline src/storage.ts:31]: I'm not sure this is correct phrase. I'd prefer to say either `Error in Storage.method()` or `Error during the call of Storage.method()`

File: src/normalize-errors-decorator.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+/* eslint no-param-reassign: "warn" */
+import 'reflect-metadata';
+import * as t from 'io-ts';
+import { StorageError } from './errors';
+
+function normalizeErrors(errorMessagePrefix = 'Error during ') {
```
Comment (artemgolozubov) [inline src/normalize-errors-decorator.ts:6]: The same here, I'm not sure this prefix is correct

File: src/storage.ts [hunk 3]
```diff
@@ -161,24 +165,23 @@ class Storage {
...
   @validate(CountryCodeIO, StorageRecordIO)
+  @normalizeErrors(STORAGE_METHOD_ERROR)
   async write(countryCode: string, record: StorageRecord, requestOptions: RequestOptions = {}): Promise<WriteResult> {
     const data = await this.encryptPayload(record);
     await this.apiClient.write(countryCode, data, requestOptions);
     return { record };
   }
 
   @validate(CountryCodeIO, StorageRecordsNEAIO)
+  @normalizeErrors(STORAGE_METHOD_ERROR)
```
Comment (artemgolozubov) [inline src/storage.ts:176]: Maybe something like `@catchUnhandledErrors()` will be more readable?

File: src/storage.ts (hunk 1 above)
Comment (igor-alekseev-dev) [reply src/storage.ts:31]: I ok with any message. @Rommero let's choose one

Comment (igor-alekseev-dev) [reply src/storage.ts:31]: I used python as an example https://github.com/incountry/sdk-python/blob/develop/incountry/validation/validate_model.py#L39

Comment (igor-alekseev-dev) [reply src/storage.ts:31]: Let's go with `Error in Storage.method()`

File: src/storage.ts (hunk 3 above)
Comment (igor-alekseev-dev) [reply src/storage.ts:176]: Unfortunately, It doesn't catch anything - only make errors which are based not on a StorageError to be StorageError

File: tests/unit/storage-tests.js [hunk 4]
```diff
@@ -636,6 +636,35 @@ describe('Storage', () => {
...
+                isForCustomEncryption: true,
+              },
+            ],
+            currentVersion: 0,
+          };
+
+          const key = '123';
+
+          const customEncConfigs = [{
+            encrypt: () => { throw new Error('blabla'); },
```
Comment (RommeroInCountry) [inline tests/unit/storage-tests.js:658]: I'd personally wrap custom enc/dec in try/catch to throw specific exceptions there :) (as a part of another PR)

File: src/storage.ts (hunk 1 above)
Comment (RommeroInCountry) [inline src/storage.ts:31]: STORAGE_METHOD_ERROR_PREFIX i'd say. Otherwise the name is misleading

**Pull request merged**